hangman/game.py

```diff
@@ -1,23 +1,70 @@
 from .exceptions import *
+import random
 
 # Complete with your own, just for fun :)
 LIST_OF_WORDS = []
 
-
 def _get_random_word(list_of_words):
-    pass
-
+    if list_of_words == []:
+        raise InvalidListOfWordsException
+    else:
+        return random.choice(list_of_words)
+    
 
 def _mask_word(word):
-    pass
+    if len(word) == 0:
+        raise InvalidWordException('Those words cannot.')
+    
+    word_length = len(word)
+    masked_word = word.replace(word, "*"*word_length)
+    return(masked_word)
 
 
 def _uncover_word(answer_word, masked_word, character):
-    pass
+    if len(answer_word) == 0 or len(masked_word) == 0:
+        raise InvalidWordException('Those words are invalid.')
+    
+    if len(character) > 1:
+        raise InvalidGuessedLetterException('Those guess letters are invalid.')
+
+    if len(answer_word) != len(masked_word):
+        raise InvalidWordException('Those words are invalid.')
 
+    result = ''
+    index = 0
+    for i in answer_word:
+        if (i.lower()) == (character.lower()):
+            result += (character.lower())
+        else:
+            result += masked_word[index]
+        index += 1
+    final_result = ''.join(result)
+    return(final_result)
+        
 
 def guess_letter(game, letter):
-    pass
+    if game['masked_word'] == game['answer_word'] or game['remaining_misses'] == 0:
+        raise GameFinishedException('The game has finished.')
+    
+    from game import _uncover_word
+    
+    if (letter.lower()) in ((game['answer_word']).lower()):
+        new_masked_word = _uncover_word(game['answer_word'], game['masked_word'], letter)
+        game.update(masked_word = new_masked_word)
```

Comment:
hmmm what are we trying to here? is it game['masked_word'] = new_masked_word ?

typically, we don't use need to use "update" method ...